submit/submit

```diff
@@ -0,0 +1,534 @@
+parser.add_argument('-h', '--help', help='display this help and exit', action='store_true')
+parser.add_argument('-c', '--contest', help='''submit for contest with ID or short name CONTEST.
+    Defaults to the value of the
+    environment variable 'SUBMITCONTEST'.
+    Mandatory when more than one contest is active.''')
+parser.add_argument('-p', '--problem', help='submit for problem with ID or label PROBLEM')
+parser.add_argument('-l', '--language', help='submit in language with ID LANGUAGE')
+parser.add_argument('-e', '--entry_point', help='set an explicit entry_point, e.g. the java main class')
+parser.add_argument('-v', '--verbose', help='increase verbosity', choices=loglevels.keys(), nargs='?', const='INFO', default='WARNING')
+parser.add_argument('-q', '--quiet', help='suppress warning/info messages', action='store_true')
+parser.add_argument('-y', '--assume-yes', help='suppress user input and assume yes', action='store_true')
+parser.add_argument('-u', '--url', help='''submit to server with base address URL
+    (should not be necessary for normal use)''')
+parser.add_argument('filename', nargs='*', help='filename(s) to submit')
+
+args = parser.parse_args()
+
+verbosity = args.verbose
+if args.quiet:
+    verbosity = 'ERROR'
+
+logging.basicConfig(format='[%(asctime)s] {}[{}]: %(message)s'.format(sys.argv[0], os.getpid()), datefmt='%b %d %H:%M:%S', level=loglevels[verbosity])
+
+contest_id = ''
+
+if 'SUBMITBASEURL' in os.environ:
+    baseurl = os.environ['SUBMITBASEURL']
+if 'SUBMITCONTEST' in os.environ:
+    contest_id = os.environ['SUBMITCONTEST']
+
+if args.contest is not None:
+    contest_id = args.contest
+problem_id = args.problem or ''
+language_id = args.language or ''
+entry_point = args.entry_point
+if args.url:
+    baseurl = args.url
+
+logging.info('set verbosity to {}'.format(verbosity))
+
+# Make sure that baseurl terminates with a '/' for later concatenation.
+if baseurl != '' and baseurl[-1:] != '/':
+    baseurl += '/'
+
+contests = read_contests()
+if contests is None:
+    logging.warning('could not obtain active contests')
+
+my_contest = None
+my_language = None
+my_problem = None
+
+if contest_id == '':
+    if contests is None or len(contests) == 0:
+        warn_user('no active contests found (and no contest specified)')
+    elif len(contests) == 1:
+        my_contest = contests[0]
+    else:
+        warn_user('multiple active contests found, please specify one')
+elif contests is not None:
+    contest_id = contest_id.lower()
+    for contest in contests:
+        if contest['id'].lower() == contest_id or contest['shortname'].lower() == contest_id:
+            my_contest = contest
+            break
+
+languages = None
+problems = None
+if my_contest is not None:
+    languages = read_languages()
+    problems = read_problems()
+
+parser.epilog = get_epilog()
+
+if args.help:
+    parser.print_help()
+    exit(0)
+
+if baseurl == '':
+    usage('no url specified')
+
+if my_contest is None:
```

Comment:
Move up to the opposite check

Reply:
Same as above for the epilog. We need get_epilog before this call, but currently need get_epilog AFTER reading problems and languages. Any suggestion?